### PR TITLE
Implemented IsTurnableEdge

### PR DESCRIPTION
**Pull request by ReymondAkpanya**

### CODE REVIEW

File: gap/PolygonalComplexes/properties.gi [hunk 1]
```diff
@@ -612,6 +612,22 @@ InstallMethod( IsInnerEdge, "for a twisted polygonal complex and an edge",
...
+	voe2:=Union(VerticesOfFaces(surface){foe});
+	voe2:=Difference(voe2,voe1);
+	if voe2 in VerticesOfEdges(surface) then 
+		return false;
+	else	
+		return true;
+	fi;
+    end
+);
 #TODO Implication to IsClosedSurface
```
Comment (TomGoertzen) [inline gap/PolygonalComplexes/properties.gi:631]: This line should come after 614 or could be deleted

File: gap/PolygonalComplexes/properties.gd [hunk 2]
```diff
@@ -457,6 +457,27 @@ DeclareOperation( "IsInnerEdge", [IsTwistedPolygonalComplex, IsPosInt] );
 DeclareOperation( "IsInnerEdgeNC", [IsTwistedPolygonalComplex, IsPosInt] );
 #! @EndGroup
 
+#! @BeginGroup IsTurnableEdge
+#! @Description
+#! Checks whether an edge turn can be performed with a given edge a
+#! simplicial surface. The function returns <A>false</A> if there exists an
+#! edge that is incident to the wing tips of the butterfly induced by
+#! <A>edge</A>, otherwise it returns <A>true</A>. The wing tips of a
```
Comment (TomGoertzen) [inline gap/PolygonalComplexes/properties.gd:465]: Isn't this true for the janus head then?

**Review by TomGoertzen — CHANGES_REQUESTED** [review]

File: gap/PolygonalComplexes/properties.gd [hunk 3]
```diff
@@ -457,6 +457,27 @@ DeclareOperation( "IsInnerEdge", [IsTwistedPolygonalComplex, IsPosInt] );
...
+#! @BeginExampleSession
+#! gap> IsTurnableEdge(Tetrahedron(),1);
+#! false
+#! gap> IsTurnableEdge(Octahedron(),2);
+#! true
+#! @EndExampleSession
+#!
+#! @Returns true or false
+#! @Arguments surface,edge
+DeclareOperation( "IsTurnableEdge", [IsPolygonalComplex, IsPosInt] );
```
Comment (MeikeWeiss) [inline gap/PolygonalComplexes/properties.gd:478]: I think it has to be DeclareProperty and then you don't need the line with the return arguments.

File: gap/PolygonalComplexes/properties.gi [hunk 4]
```diff
@@ -612,6 +612,22 @@ InstallMethod( IsInnerEdge, "for a twisted polygonal complex and an edge",
...
+    function(surface,edge)
+	local g,voe1,voe2,foe;
+	voe1:=VerticesOfEdge(surface,edge);
+	foe:=FacesOfEdge(surface,edge);
+	voe2:=Union(VerticesOfFaces(surface){foe});
+	voe2:=Difference(voe2,voe1);
+	if voe2 in VerticesOfEdges(surface) then 
+		return false;
+	else	
+		return true;
```
Comment (MeikeWeiss) [inline gap/PolygonalComplexes/properties.gi:627]: In this way the method returns true for a boundary edge. So, there has to be a query if edge is an inner edge or not.

File: gap/PolygonalComplexes/properties.gd (hunk 3 above)
Comment (MeikeWeiss) [reply gap/PolygonalComplexes/properties.gd:493]: This is a good point. But then also a method TurnableEdges and a NC-version would make sense.

**Review by TomGoertzen — APPROVED** [review]
Looks good :)

File: gap/PolygonalComplexes/properties.gd [hunk 5]
```diff
@@ -457,6 +457,54 @@ DeclareOperation( "IsInnerEdge", [IsTwistedPolygonalComplex, IsPosInt] );
 DeclareOperation( "IsInnerEdgeNC", [IsTwistedPolygonalComplex, IsPosInt] );
 #! @EndGroup
 
+#! @BeginGroup TurnableEdges
+#! @Description
+#! Return the set of all turnable edges of the given simplicial surface.
+#! Let <A>v1,v2</A> be the vertices of the subsurface
+#! induced by a given edge which are not incident this edge.
```
Comment (MeikeWeiss) [inline gap/PolygonalComplexes/properties.gd:464]: Here is a "to" missing.

File: gap/PolygonalComplexes/properties.gd [hunk 6]
```diff
@@ -457,6 +457,54 @@ DeclareOperation( "IsInnerEdge", [IsTwistedPolygonalComplex, IsPosInt] );
...
+#! <Alt Only="TikZ">
+#!     \input{Image_ButIndEdg.tex}
+#! </Alt>
+#! The given edge is a turnable edge, if and only if <A>v1</A> is not equal to
+#! <A>v2</A> and there exist no edge whose incident vertices are exactly
+#! <A>v1</A> and <A>v2</A>.
+#! The method IsTurnableEdge checks whether the given edge is a turnable edge
```
Comment (MeikeWeiss) [inline gap/PolygonalComplexes/properties.gd:471]: It would be better to write IsTurnableEdge with < A > and < \A >.

**Review by MeikeWeiss — CHANGES_REQUESTED** [review]
Looks nice!

**Review by MeikeWeiss — APPROVED** [review]
Nice!

**Review by TomGoertzen — APPROVED** [review]
Looks fine now!

**Pull request merged**